diesel/src/query_builder/query_id.rs

```diff
@@ -1,11 +1,54 @@
 use std::any::{Any, TypeId};
 use super::QueryFragment;
 
+/// Uniquely identifies queries by their type for the purpose of prepared
+/// statement caching.
+///
+/// All types which implement `QueryFragment` should also implement this trait
+/// (It is not an actual supertrait of `QueryFragment` for boxing purposes).
+///
+/// This trait is generally implemented by invoking the [`impl_query_id!`]
```

Comment:
As written above: If we want to replace this macro, we should not mention it in the documentation.